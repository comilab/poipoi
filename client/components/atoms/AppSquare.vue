<template lang="pug">
.is-square.relative
  .content.absolute.inset-0
    slot
</template>

<script lang="ts">
import { defineComponent } from 'nuxt-composition-api'

export default defineComponent({})
</script>

<style lang="postcss" scoped>
.is-square {
  padding-top: 100%;

  & .content > * {
    @apply w-full h-full;
  }
}
</style>
