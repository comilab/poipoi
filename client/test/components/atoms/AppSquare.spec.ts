import { render, RenderOptions } from '@testing-library/vue'

import AppSquare from '~/components/atoms/AppSquare.vue'

describe('components/atoms/AppSquare', () => {
  let options: RenderOptions<AppSquare>

  beforeEach(() => {
    options = {
      slots: {
        default: 'default-slot'
      }
    }
  })

  test('renders correctly', () => {
    const { container, getByText } = render(AppSquare, options)

    expect(container).toMatchSnapshot()
    getByText('default-slot')
  })
})
