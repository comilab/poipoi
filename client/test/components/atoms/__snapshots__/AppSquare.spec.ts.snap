// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/AppSquare renders correctly 1`] = `
<div>
  <div
    class="is-square relative"
  >
    <div
      class="content absolute inset-0"
    >
      default-slot
    </div>
  </div>
</div>
`;
